config/knative/addressables.duck.knative.dev.yaml

```diff
@@ -21,7 +21,10 @@ metadata:
 spec:
   selectors:
     - labelSelector: "duck.knative.dev/addressable=true"
-
+  clusterRoleRef:
```

Comment:
I am trying to decide if this shape or something like this makes sense:

```
clusterRole:
  ref:
    name,kind,apiGroup: ...
  selector: "duck.knative.dev/addressable=true"
```

and if the selector here is used, it use this to find out how to select the role. Or defaults to `spec.selectors.labelSelector`

Reply:
The defaulting idea sounds good. Updated the functionality so it defaults to `spec.selectors.labelSelector`.

@n3wscott  Do you feel adding the `clusterRole.selector` also would be useful ?

Reply:
Yeah, I think so long term, but we should just make room for it in this PR and not implement it yet.

Reply:
sounds good. I'll move the clusterRoleRef inside a struct to make room for the selector in the future